### PR TITLE
feat(vite): experimental `vite-node` support

### LINKED ISSUE #645: Vite dev sourcemap support
Followup #604

### PR DESCRIPTION
**Pull request by antfu**
Close #645

### ❓ Type of change

- [ ] 📖 Documentation (updates to the documentation or readme)
- [ ] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [x] 👌 Enhancement (improving an existing functionality like performance)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📝 Changes

- Now we no longer need the dev bundler and no need to bundle `server.mjs` anymore
- And we also don't need the watcher since everything is fetched on-demand (via HTTP requests)
- Client manifests are generated from `server.mjs` which is from the requested module tree, and also on-demand
- Now we could have stacktraces for server errors with correct sourcemap :)

### TODO

- [ ] Do a measurement and compare with a Nuxt 3 codebase to see the improvements in numbers
- [ ] Migrate for bridge (in another PR)

### CODE REVIEW

File: packages/vite/src/vite-node.ts [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+import { ViteNodeServer } from 'vite-node/server'
+import fse from 'fs-extra'
+import { resolve } from 'pathe'
+import { addServerMiddleware } from '@nuxt/kit'
+import { distDir } from './dirs'
+import type { ViteBuildContext } from './vite'
+
+export function registerViteNodeMiddleware (ctx: ViteBuildContext) {
+  let viteNode: ViteNodeServer | undefined
+  addServerMiddleware({
```
Comment (antfu) [inline packages/vite/src/vite-node.ts:11]: @pi0 the server middleware seems not been registered, did I missed anything?

**Comment by ayalon** [comment]
Will this merge request enable source maps for scss files using vite and nuxt?

**Comment by misaon** [comment]
> Will this merge request enable source maps for scss files using vite and nuxt?

I do not think so

**Review by pi0 — APPROVED** [review]

**Comment by pi0** [comment]
Next steps: https://github.com/nuxt/framework/issues/3598

**Pull request merged**